(4.x.x) Some general code cleanup

adamretter opened the pull request:
Backport of https://github.com/eXist-db/exist/pull/2105

adamretter commented:
@dizzzz Ah I missed a small think in the rebase. Should be good now...

dizzzz commented on bin/functions.d/eXist-settings.sh:
```diff
@@ -1,6 +1,8 @@
-# -*-Shell-script-*-
-# Common eXist script functions and settings
-# $Id:eXist-settings.sh 7231 2008-01-14 22:33:35Z wolfgang_m $
+#!/usr/bin/env bash
```
A wise (?) man (you?) told me once that is this is a dangerous construct... makes the script vulnerable ?

adamretter replied:
Hmmm. I am not certain about that.

However it seems that these days, `#!/usr/bin/env bash` is recommended over `#!/bin/bash` for portability (not all systems have bash in `/bin` e.g. NixOS).

adamretter commented:
@dizzzz Rebased as requested

dizzzz commented:
:-( the process is really silent for 10+ minutes. De we have a deadlock ?

adamretter commented:
@dizzzz It was a dangling lock problem.
Somehow whilst merging during the back-port I removed a line that released a lock. I have now reinstated that, rebased and pushed.

All should be good now :-)
